qoqo_braket/backend.py

```diff
@@ -0,0 +1,290 @@
+    ) -> None:
+        """Initialise the BraketBackend class.
+
+        Args:
+            device: Optional ARN of the Braket device to use. If none is provided, the
+                    default LocalSimulator will be used.
+            aws_session: Optional AwsSession to use. If none is provided, a new one will be created
+            verbatim_mode: Whether to use verbatim boxes to avoid recompilation
+        """
+        self.aws_session = aws_session
+        self.device = "braket_sv" if device is None else device
+        self.verbatim_mode = verbatim_mode
+
+        self.__use_actual_hardware = False
+        self.__force_rigetti_verbatim = False
+        self.__force_ionq_verbatim = False
+        self.__force_oqc_verbatim = False
+        self.__max_circuit_length = 100
+        self.__max_number_shots = 100
+
+    def allow_use_actual_hardware(self) -> None:
+        """Allow the use of actual hardware - will cost money."""
+        self.__use_actual_hardware = True
+
+    def disallow_use_actual_hardware(self) -> None:
+        """Disallow the use of actual hardware."""
+        self.__use_actual_hardware = False
+
+    def force_rigetti_verbatim(self) -> None:
+        """Force the use of rigetti verbatim. Mostly used for testing purposes."""
+        self.__force_rigetti_verbatim = True
+
+    def force_ionq_verbatim(self) -> None:
+        """Force the use of ionq verbatim. Mostly used for testing purposes."""
+        self.__force_ionq_verbatim = True
+
+    def force_oqc_verbatim(self) -> None:
+        """Force the use of oqc verbatim. Mostly used for testing purposes."""
+        self.__force_oqc_verbatim = True
+
+    def change_max_shots(self, shots: int) -> None:
+        """Change the maximum number of shots allowed.
+
+        Args:
+            shots: new maximum allowed number of shots
+        """
+        self.__max_number_shots = shots
+
+    def change_max_circuit_length(self, length: int) -> None:
+        """Change the maximum circuit length allowed.
+
+        Args:
+            length: new maximum allowed length of circuit
+        """
+        self.__max_circuit_length = length
+
+    def __create_device(self) -> Union[LocalSimulator, AwsDevice]:
+        """Creates the device and returns it.
+
+        Returns:
+            The instanciated device (either an AwsDevice or a LocalSimulator)
+
+        Raises:
+            ValueError: Device specified isn't allowed
+        """
+        if self.device.startswith("local:") or self.device in LOCAL_SIMULATORS_LIST:
+            device = LocalSimulator(self.device)
+        elif self.device in REMOTE_SIMULATORS_LIST:
+            device = AwsDevice(self.device)
+        else:
+            if self.__use_actual_hardware:
+                # allow list simulator devices of AWS e.g. state vector simulator
+                device = AwsDevice(self.device)
+            else:
+                raise ValueError("Device specified isn't allowed")
```

Comment:
We could mention that it can be explicitely allowed but there is a danger of incurring large costs in the error message.